README.md

```diff
@@ -1,2 +1,437 @@
+
+  ```js
+  import { connect } from 'react-redux';
+  import { Link } from 'react-router-dom';
+  import { Card, Button } from 'semantic-ui-react';
+  import { fetchUsers } from 'actions/users';
+  import searchBy from 'utils/search';
+
+  import Search from 'components/helpers/Search';
+  import Page from 'components/layouts/Page';
+  ```
+
+## JSX
+
+JSX leverages syntactic similarity with HTML so similar conventions should be observed.
+
+* rendering
+  ```js
+  // good
+  render() {
+    return <Button text="Save" onChange={this.save}/>;
+  }
+  // good
+  render() {
+    return (
+      <div className="action-bar">
+        <Button text="Save" onChange={this.save}/>
+        <Button text="Cancel" onChange={this.goBack}/>
+      </div>
+    );
+  }
+
+  // bad
+  render() {
+    return <div className="action-bar">
+      <Button text="Save" onChange={this.save}/>
+      <Button text="Cancel" onChange={this.goBack}/>
+    </div>;
+  }
+  ```
+
+* use double quotes for attributes
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text='Save' onChange={this.save}/>
+  ```
+
+* Do not use braces around double quotes
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text={"Save"} onChange={this.save}/>
+  ```
+
+* Do not use spaces around =
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text = "Save" onChange = {this.save}/>
+  ```
+
+* Inline short components
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button
+    text="Save"
+    onChange={this.save}/>
+  ```
+
+* Multiline each prop on it's own line
+  ```js
+  // good
+  <Input
+    type="email"
+    label="Email"
+    placeholder="name@example.com"
+    onChange={this.setValue('email')}
+    onClick={this.selectText}/>
+
+  // bad
+  <Input
+    type="email" label="Email" placeholder="name@example.com"
+    onChange={this.setValue('email')} onClick={this.selectText}/>
+  ```
+
+* Keys should come first and may be on the first line when multilined.
+  ```js
+  // ok
+  <ul>
+    {cars.map((car) => (
+      <Car key={car.id}
+        car={car}
+        detail="list"
+        showPrice={true}/>
+    ))}
+  </ul>
+
+  // bad
+  <ul>
+    {cars.map((car) => (
+      <Car
+        car={car}
+        detail="list"
+        key={car.id}
+        showPrice={true}/>
+    ))}
+  </ul>
+  ```
+
+* Multiline contents if there are nested elements.
+  ```js
+  // good
+  <p>Hi {name}, welcome to the site!</p>
+  // good
+  <p>
+    <i className="icon-checkmark"/>
+    <strong>Hi {name}</strong>, welcome to the site!
+  </p>
+
+  // bad
+  <p><i className="icon-checkmark"/><strong>Hi {name}</strong>, welcome to the site!</p>
+  ```
+
+* Extract components, don't have multiple rendering functions. This will make the class shorter and more readable and make it easier to extract and reuse common components.
```

Comment:
Maybe we can discuss this a bit more.

I am specifically trying to say that both styles of "extraction" in the bad example are bad.

For `renderErrorMessage`, it is bad because:
- It makes it hard to reuse in other components.
- It clutters up the class. It makes it longer and harder to scan.
- It obscures the dependencies - specifically, it only relates to a single value: `props.error`. We could destructure assignment to make this more clear but at this point we've made it longer than the "good" example.

For `showCars` it is bad because we've broken up the render function arbitrarily. The `showCars` function doesn't quite do enough to stand alone. I have to bounce around the class to figure out what actually gets rendered. If we wanted to reuse a "CarsList" component maybe we would exact a full component and use JSX because that's what JSX if for (but this component is already a CarsList Component so...).